src/fastertransformer/triton_backend/llama/LlamaTritonModelInstance.h

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2022-2023, NVIDIA CORPORATION.  All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "src/fastertransformer/models/llama/Llama.h"
+#include "src/fastertransformer/triton_backend/llama/LlamaTritonModel.h"
+#include "src/fastertransformer/triton_backend/transformer_triton_backend.hpp"
+#include <memory>
+
+namespace ft = fastertransformer;
+
+template<typename T>
+struct LlamaTritonModelInstance: AbstractTransformerModelInstance {
+
+    LlamaTritonModelInstance(std::unique_ptr<ft::Llama<T>>                           gpt,
+                             std::shared_ptr<ft::LlamaWeight<T>>                     gpt_weight,
+                             std::unique_ptr<ft::Allocator<ft::AllocatorType::CUDA>> allocator,
+                             std::unique_ptr<ft::cublasAlgoMap>                      cublas_algo_map,
+                             std::unique_ptr<std::mutex>                             cublas_wrapper_mutex,
+                             std::unique_ptr<ft::cublasMMWrapper>                    cublas_wrapper,
+                             std::unique_ptr<cudaDeviceProp>                         cuda_device_prop_ptr);
+    ~LlamaTritonModelInstance();
+
+    std::shared_ptr<std::vector<triton::Tensor>>
+    forward(std::shared_ptr<std::vector<triton::Tensor>> input_tensors) override;
+
+    std::shared_ptr<std::unordered_map<std::string, triton::Tensor>>
+    forward(std::shared_ptr<std::unordered_map<std::string, triton::Tensor>> input_tensors) override;
+
+    static std::shared_ptr<std::unordered_map<std::string, triton::Tensor>>
+    convert_outputs(const std::unordered_map<std::string, ft::Tensor>& output_tensors);
+
+private:
+    const std::unique_ptr<ft::Llama<T>>                           gpt_;
+    const std::shared_ptr<ft::LlamaWeight<T>>                     gpt_weight_;
+    const std::unique_ptr<ft::Allocator<ft::AllocatorType::CUDA>> allocator_;
```

Comment:
you're gonna want to swap this so things destruct correctly

```suggestion
    const std::unique_ptr<ft::Allocator<ft::AllocatorType::CUDA>> allocator_;
    const std::unique_ptr<ft::Llama<T>>                           gpt_;
    const std::shared_ptr<ft::LlamaWeight<T>>                     gpt_weight_;
```